### PR TITLE
Add TimeoutingSampler module and utils package and use it

### PR DESCRIPTION
**Pull request by KwisatzHaderach**

### CODE REVIEW

File: ovirt_sdk_helpers/tests/test_hosts.py [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import pytest
```
Comment (StLuke) [inline ovirt_sdk_helpers/tests/test_hosts.py]: dont think we should include this test

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 2]
```diff
@@ -88,7 +97,7 @@ def upgrade_host(host_name, image=None, async=False, reboot=True):
     """
     host = get_host_by_name(host_name)
     host_service = (
-        connection._CON.system_service().hosts_service().host_service(host.id)
+        osh.connection._CON.system_service().hosts_service().host_service(host.id)
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: s/_CON/con

Comment (StLuke) [reply ovirt_sdk_helpers/sdk/hosts.py]: s/create_conenction/init/

**Review by StLuke — CHANGES_REQUESTED** [review]

**Review by StLuke — CHANGES_REQUESTED** [review]
remove all tests for now

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 3]
```diff
@@ -35,6 +33,18 @@ def is_upgrade_available(host_name):
     return get_host_by_name(host_name).update_available
 
 
+def _wait_for_update_check(host_name, events_service, last_event_id):
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: docstring

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 4]
```diff
@@ -35,6 +33,18 @@ def is_upgrade_available(host_name):
...
+        search="host.name={}".format(host_name)
+    ):
+        if event.description.startswith(HOST_UPGRADE_CHECK_FINISHED):
+            return True
+        if event.description.startswith(HOST_UPGRADE_CHECK_FAILED):
+            return False
+    return None
+
+
 def check_host_upgrade(host_name, wait=False):
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: s/check_host_upgrade/check_upgrade/

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 5]
```diff
@@ -45,31 +55,32 @@ def check_host_upgrade(host_name, wait=False):
...
+        )
     )
 
     if not host.update_available:
         host_service.upgrade_check()
 
-    timeout = 0
     if wait:
-        events_service = connection._CON.system_service().events_service()
+        events_service = osh.connection.con.system_service().events_service()
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: move this to events.py

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 6]
```diff
@@ -45,31 +55,32 @@ def check_host_upgrade(host_name, wait=False):
...
         bool: True if action succeeds otherwise False
+
+    Throws:
+        TimeoutExpiredError: when timeout expires, neither succeeded nor
+        failed update check message appeared
     """
     host = get_host_by_name(host_name)
     host_service = (
-        connection._CON.system_service().hosts_service().host_service(host.id)
+        osh.connection.con.system_service().hosts_service().host_service(
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: def service():
    return con.system_service.hosts_service

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 7]
```diff
@@ -18,7 +16,7 @@ def get_host_by_name(host_name):
     Returns:
         ovirtsdk4.types.Host: host object
     """
-    hosts_service = connection._CON.system_service().hosts_service()
+    hosts_service = osh.connection.con.system_service().hosts_service()
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: s/hosts_service/

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 8]
```diff
@@ -88,7 +99,9 @@ def upgrade_host(host_name, image=None, async=False, reboot=True):
     """
     host = get_host_by_name(host_name)
     host_service = (
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: def service()
   return connection...host_service
service = service()

File: ovirt_sdk_helpers/sdk/hosts.py [hunk 9]
```diff
@@ -45,31 +55,32 @@ def check_host_upgrade(host_name, wait=False):
...
-        return False
+        for sample in osh.utils.TimeoutingSampler(
+            CHECK_UPGRADE_TIMEOUT, CHECK_UPGRADE_SLEEP, _wait_for_update_check,
+            host.name, events_service, last_event.id
+        ):
+            if sample:
+                return True
+            if not sample and sample is not None:
+                return False
     return True
```
Comment (StLuke) [inline ovirt_sdk_helpers/sdk/hosts.py]: s/upgrade_host/upgrade/

just Ctrl+F host in host.py

**Review by StLuke — CHANGES_REQUESTED** [review]

**Review by StLuke — APPROVED** [review]

**Review by pkubica — APPROVED** [review]

**Pull request merged**